tensorflow_model_optimization/python/core/sparsity/keras/prune_integration_test.py

```diff
@@ -126,8 +126,8 @@ def setUp(self):
   # these checks are redundant.
   def _check_strip_pruning_matches_original(
       self, model, sparsity, input_data=None):
+    test_utils.assert_model_sparsity(self, sparsity, model)
     stripped_model = prune.strip_pruning(model)
-    test_utils.assert_model_sparsity(self, sparsity, stripped_model)
```

Comment:
Yes (#631 ):

> This [specific function](https://github.com/tensorflow/model-optimization/blob/4f4e5d02db5424d5cfcb822a8965023bf8d2011e/tensorflow_model_optimization/python/core/sparsity/keras/prune_integration_test.py#L127) doesn't work the way intended, because:
> 
> - The model is first stripped, so there are no more pruning wrappers in the model.
> - Then, the model is given to [assert_model_sparsity](https://github.com/tensorflow/model-optimization/blob/4f4e5d02db5424d5cfcb822a8965023bf8d2011e/tensorflow_model_optimization/python/core/sparsity/keras/test_utils.py#L167).
> - However, this last function only asserts layers that have a pruning wrapper. Thus, the sparsity is not checked at all.

I understand that it is intended to check the sparsity after pruning wrappers are stripped. However, assert_model_sparsity doesn't assert the sparsity of non-pruning wrappers, thus it has no effect at all.

My fix allows to at least check the sparsity correctly (before pruning however...). I understand if the proposed fix is not fully in line with the expected behavior. If it is absolutely necessary to check the sparsity after stripping the behavior of assert_model_sparisity have to be modified though.